service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java

```diff
@@ -626,17 +626,15 @@ public ResponseEntity<ApiCloneWorkspaceResult> cloneWorkspace(
     //  this part, we do the retrieval and permission check using the workspace spend profile.
     Optional<SpendProfileId> spendProfileIdOptional =
         Optional.ofNullable(body.getSpendProfile()).map(SpendProfileId::new);
-    SpendProfileId spendProfileId =
-        spendProfileIdOptional.orElse(
-            sourceWorkspace
-                .getSpendProfileId()
-                .orElseThrow(
-                    () ->
-                        MissingSpendProfileException.forWorkspace(sourceWorkspace.workspaceId())));
 
-    SpendProfile spendProfile =
-        spendProfileService.authorizeLinking(
-            spendProfileId, features.isBpmGcpEnabled(), petRequest);
+    if (spendProfileIdOptional.isEmpty()) {
+      spendProfileIdOptional = sourceWorkspace.getSpendProfileId();
+    }
+    Optional<SpendProfile> spendProfileOptional =
+        spendProfileIdOptional.map(
+            spendProfileId ->
+                spendProfileService.authorizeLinking(
+                    spendProfileId, features.isBpmGcpEnabled(), petRequest));
```

Comment:
I like Optionals, especially for defining default values inline, and I see that they are widely used through the repo. But sometimes I find that they can obscure the intent of a block of code, which is the case for me here.

The intent is to fall back to the source workspace's spend profile if none is specified.  We are okay with passing a null profile ID to the `Workspace.Builder` and a null profile to `workspaceservice.cloneWorkspace`.

I found the following to better convey the intent, and then you don't have to unpack the Optionals further down in the method.  Thoughts?

```
    SpendProfileId workspaceSpendProfileId = sourceWorkspace.getSpendProfileId().orElse(null);
    SpendProfileId spendProfileId =
            Optional.ofNullable(body.getSpendProfile()) 
                    .map(SpendProfileId::new)
                    .orElse(workspaceSpendProfileId);
    SpendProfile spendProfile = null;
    if (spendProfileId != null) {
      spendProfile = spendProfileService.authorizeLinking(spendProfileId, features.isBpmGcpEnabled(), petRequest);
    }
```

Reply:
I like it, I felt this was all a bit clunky so thank you for the suggestion! I'm very accustomed to using Scala Options and avoiding null at all costs, but I haven't found Java Optionals to be quite as nice.